src/components/CartPage/CartItem/CartItem.js

```diff
@@ -7,94 +7,165 @@ import './CartItem.scss';
+          >
+            <i className="icon close small" />
+          </button>
+        </div>
+      </div>
+    );
   };
 
-  const handleQuantity = () => ((chosenQuantity === 1) ? onRemoveFromCart : () => changeAmount(dec));
-
-  const renderProductDescription = () => (
-    <div className={`${CN}__description`}>
-      <img className="col-3" src={`${process.env.IMAGE_URL}/${images[0].claudinaryId}`} alt={title} />
-      <div className={`${CN}__text col-8 col-right`}>
-        <h4 className={`${CN}__title`}>{title}</h4>
-        <h4>
-          {`color : ${color}`}
-        </h4>
-        <h4>
-          {`size: ${size}`}
-        </h4>
-        <Link to={`/product-details/${_id}`} target="_blank">
-          <p className={`${CN}__edit`}>
-          Edit item
-          </p>
-        </Link>
+  renderCounter = () => {
+    const { item, userCart } = this.props;
+    const { quantity, _id } = item;
+    const userCartItem = userCart.find((cartItem) => cartItem.id === _id) || {};
+    const { chosenQuantity } = userCartItem;
+    const { inc } = this.state;
+
+    return (
+      <div className={`${CN}__counter`}>
+        <button
+          className={`${CN}__button`}
+          type="button"
+          onClick={this.handleQuantity()}
+        >
+          <i className="small minus icon" />
+        </button>
+        <p className={`${CN}__quantity`}>{chosenQuantity}</p>
         <button
           className={`${CN}__button`}
           type="button"
-          onClick={onRemoveFromCart}
+          onClick={() => this.changeAmount(inc)}
+          disabled={chosenQuantity === quantity}
         >
-          <i className="icon close small" />
+          <i className="small plus icon" />
         </button>
       </div>
-    </div>
-  );
-
-  const renderCounter = () => (
-    <div className={`${CN}__counter`}>
-      <button
-        className={`${CN}__button`}
-        type="button"
-        onClick={handleQuantity()}
-      >
-        <i className="small minus icon" />
-      </button>
-      <p className={`${CN}__quantity`}>{chosenQuantity}</p>
-      <button
-        className={`${CN}__button`}
-        type="button"
-        onClick={() => changeAmount(inc)}
-        disabled={chosenQuantity === quantity}
-      >
-        <i className="small plus icon" />
-      </button>
-    </div>
-  );
-
-  return (
-    <tr className={cx(CN, className)} key={title}>
-      <td className={`${CN}__column`}>{renderProductDescription()}</td>
-      <td className={`${CN}__column`}>{`$${price}`}</td>
-      <td className={`${CN}__column`}>{renderCounter()}</td>
-      <td className={`${CN}__column`}>{`$${price * chosenQuantity}`}</td>
-    </tr>
-  );
-};
+    );
+  };
+
+  render() {
+    const { convertedPrice } = this.state;
+    const {
+      item,
+      userCart,
+      className,
+      symbol
+    } = this.props;
+    const { _id, title } = item;
+    const userCartItem = userCart.find((cartItem) => cartItem.id === _id) || {};
```

Comment:
If this you have to use **userCartItem** so badly, why not set it in 1 place like **constructor** and have a reference to it through this.userCartItem. Instead of defining it in 4 different methods which is not very good for performance reasons